py-polars/polars/dataframe/frame.py

```diff
@@ -3172,13 +3175,20 @@ def write_database(
                 ) from exc
 
             engine_sa = create_engine(connection_uri)
+            db_schema = None
+            if not table_name.startswith('"') and "." in table_name:
+                db_schema, table_name = table_name.split(".", 1)
```

Comment:
Should this be `rsplit(".", 1)` instead? I imagine a scenario where the schema contains `.` eg if it is `database.schema`. (Not sure if that's valid in syntax in any DB)

Reply:
The database is supplied with the URI rather than the table name.

Reply:
Aha! Thought of an edge-case overnight... if somebody quote the schema separately we'd miss it. Will update accordingly 👍